examples/custom-environment/lib.rs

```diff
@@ -0,0 +1,132 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink::env::Environment;
+
+/// Our custom environment diverges from the `DefaultEnvironment` in the event topics limit.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum EnvironmentWithManyTopics {}
+
+impl Environment for EnvironmentWithManyTopics {
+    // We allow for 5 topics in the event, therefore the contract pallet's schedule must allow for
+    // 6 of them (to allow the implicit topic for the event signature).
+    const MAX_EVENT_TOPICS: usize =
+        <ink::env::DefaultEnvironment as Environment>::MAX_EVENT_TOPICS + 1;
+
+    type AccountId = <ink::env::DefaultEnvironment as Environment>::AccountId;
+    type Balance = <ink::env::DefaultEnvironment as Environment>::Balance;
+    type Hash = <ink::env::DefaultEnvironment as Environment>::Hash;
+    type BlockNumber = <ink::env::DefaultEnvironment as Environment>::BlockNumber;
+    type Timestamp = <ink::env::DefaultEnvironment as Environment>::Timestamp;
+
+    type ChainExtension = <ink::env::DefaultEnvironment as Environment>::ChainExtension;
+}
+
+#[ink::contract(env = crate::EnvironmentWithManyTopics)]
+mod runtime_call {
+    /// Trivial contract with a single message that emits an event with many topics.
+    #[ink(storage)]
+    #[derive(Default)]
+    pub struct Topicer;
+
+    /// An event that would be forbidden in the default environment, but is completely valid in
+    /// our custom one.
+    #[ink(event)]
+    #[derive(Default)]
+    pub struct TopicedEvent {
+        #[ink(topic)]
+        first_topic: Balance,
+        #[ink(topic)]
+        second_topic: Balance,
+        #[ink(topic)]
+        third_topic: Balance,
+        #[ink(topic)]
+        fourth_topic: Balance,
+        #[ink(topic)]
+        fifth_topic: Balance,
+    }
+
+    impl Topicer {
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            Self {}
+        }
+
+        /// Emit an event with many topics.
+        #[ink(message)]
+        pub fn trigger(&mut self) {
+            self.env().emit_event(TopicedEvent::default());
+        }
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+
+        #[ink::test]
+        fn emits_event_with_many_topics() {
+            let mut contract = Topicer::new();
+            contract.trigger();
+
+            let emitted_events = ink::env::test::recorded_events().collect::<Vec<_>>();
+            assert_eq!(1, emitted_events.len());
+
+            let _ = <<Topicer as ink::reflect::ContractEventBase>::Type as scale::Decode>::decode(&mut &emitted_events[0].data[..])
+                .expect("encountered invalid contract event data buffer");
```

Comment:
What are you trying to check for here?

Reply:
if `TopicWithEvents` has been emitted

Reply:
It's just checking that _any_ even has been emitted though, no?

We should check that the type is also what we expect.